cli/server.go

```diff
@@ -1248,3 +1253,21 @@ func startBuiltinPostgres(ctx context.Context, cfg config.Root, logger slog.Logg
 	}
 	return connectionURL, ep.Stop, nil
 }
+
+func handleOauth2ClientCertificates(cfg *codersdk.DeploymentConfig, ctx context.Context) (context.Context, error) {
+	if cfg.TLS.ClientCertFile.Value != "" && cfg.TLS.ClientKeyFile.Value != "" {
```

Comment:
We already have `ClientCAFile` as a configuration option, so you should be able to remove the configuration changes here!

Reply:
I'm going to expand on my use-case a bit more because I _think_ it's slightly different

So with my changes in the config file we have:
* `CertFiles`
* `KeyFiles`
* `ClientCAFile`
* `ClientCertFile` (new)
* `ClientKeyFile` (new)

In my org, I have been given a wildcard cert `*.coder.normana10.example.com` and a "system/identity" cert `normana10.example.com`

I need to use the wildcard cert for all "serving" and the "identity" cert for all external calls _that Coder makes_

(I'm not going to say this is "correct", but it's just how my org does things)

So if I wanted Coder to terminate TLS my config would look like:
* `CertFiles` = `/path/to/wildcard.cert`
* `KeyFiles` = `/path/to/wildcard.key`
* `ClientCAFile` = `/path/to/ca.cert` (Only used to verify client certs if `ClientAuth` is set to verify)
* `ClientCertFile` (new) = `/path/to/identity.cert`
* `ClientKeyFile` (new) = `/path/to/identity.key`

So `ClientCAFile` (as far as I've seen) looks like it's just used to verify _clients connecting to Coder_. Where I'm looking to set the client certs _Coder uses when connecting to external services_

If that makes sense?

For full transparency, my _actual_ config doesn't have Coder terminating TLS, so realistically I'd _just_ have my two new configs set

Reply:
All makes sense! I appreciate the context... the changes look good to me!